src/plugins/_manager.ts

```diff
@@ -0,0 +1,171 @@
+import * as child from 'child_process';
+import * as os from 'os';
+import * as path from 'path';
+import * as url from 'url';
+import * as fs from 'fs-extra';
+
+const MODULE_NOT_FOUND_ERROR_CODE = 'MODULE_NOT_FOUND'; // TODO is there a known constant we can use here?
+const MODULE_VERSION_MISMATCH_ERROR_CODE = 'MODULE_VERSION_MISMATCH';
+
+/**
+ * Error indicating a module was found but with an unexpected version.
+ */
+export class ModuleVersionMismatchError extends Error {
+
+  public readonly code: string = MODULE_VERSION_MISMATCH_ERROR_CODE;
+
+  constructor(pkg: string, expected: string, actual: string) {
+    super(`Version mismatch for package ${pkg}. Found ${actual} but requested ${expected}`);
+  }
```

Comment:
This is fine, but you could as well:

```suggestion
  constructor(
    public readonly pkg: string,
    public readonly expected: string,
    public readonly actual: string,
  ) {
    super(`Version mismatch for package ${pkg}. Found ${actual} but requested ${expected}`);
  }
```

Reply:
👍